### ISSUE TITLE
Ipad 4 wifi "activation could not be completed" ios 8.4.1 downgrade

### ISSUE THREAD

**Issue by nizardho** [LukeZGD/iOS-OTA-Downgrader#280]
Good evening, dear LukeZGD,
I have an iPad 4 Wi-Fi with iOS 10.3.4. I tried to downgrade it to io 8.4.1. Everything went well but when at the boot I face "activation error could not be completed". I tried to restore again with nvram clearing and I got the same error activation. Here are the logs: 
[log.txt](https://github.com/LukeZGD/iOS-OTA-Downgrader/files/11011486/log.txt)
Here is the error pic
![IMG_20230316_235838](https://user-images.githubusercontent.com/81031415/226186331-2d30aa6d-83a1-474b-9424-729fbee43850.jpg)
Nice greetings,

**Comment by LukeZGD** [comment]
I don't know why that issue happens to your device, sorry. It's not really a problem with the script also since it works fine for others...

I think this Reddit comment might be able to help: https://www.reddit.com/r/jailbreak/comments/7mjc5s/news_activation_records_from_ios_10_can_activate/druvon4/

It involves backing up activation records in a tar file, then putting them in a custom IPSW. Ignore the "What you need" section, and you only need to do steps 1 to 12 to get the activation tar file. Then you need to modify `restore.sh` to use them in iOS-OTA-Downgrader:

- Replace `device_disable_bbupdate` with your device
- Place the activation records tar in the iOS-OTA-Downgrader folder
- For jailbreak option enabled: add your tar file in line 1714. Example: `"$ipsw" "$ipsw_path.ipsw" temp.ipsw $ExtraArgs ${JBFiles[@]} activation_records.tar`
- For jailbreak option disabled: add your tar file in line 1823. Example: `"$dir/powdersn0w" "$ipsw_path.ipsw" temp.ipsw $ExtraArgs activation_records.tar`

**Comment by RuZniki** [comment]
Hello thank you for very detailed description and for the great software.
I used latest Legacy iOS Kit v24.01.01 (0ba8a76), and I was able to fix this error.

On iPad I install 10.3.3 iOS, and activate it.
Run `./restore.sh` - 6) Other Utilities - 5) Activation Records. 
I get activation records in `saved/iPad3,4/activation.tar`. Good idea to copy it somewhere.
The `activation.tar` file structure is different in the instruction from reddit.
Generated Legacy iOS Kit: `/private/var/root/Library/Lockdown/activation_record.plist`
Requested from reddit: `/private/var/root/Library/Lockdown/activation_records/activation_record.plist`
I made changes to tar file to have same structure, not sure if this is necessary.

After it I used `./restore.sh --activation-records` to downgrade iOS 10.3.3 to iOS 8.4.1. 
But double check that you removed previous custom IPSW, otherwise new IPSW will not be generate and will be used already existed.
If you see message:
> Found existing Custom IPSW. Skipping IPSW creation.

then IPSW is not generated. Delete `iPad3,4_8.4.1_12H321_CustomJ.ipsw` file and start again.

In the end I was able to downgrade with JailBreak and skip activation error. I am so happy. Thanks a lot for a great software.
